src/ObjLoading/Game/IW4/AssetLoaders/AssetLoaderLeaderboard.cpp

```diff
@@ -15,3 +18,31 @@ void* AssetLoaderLeaderboard::CreateEmptyAsset(const std::string& assetName, Mem
     leaderboard->name = memory->Dup(assetName.c_str());
     return leaderboard;
 }
+
+bool AssetLoaderLeaderboard::CanLoadFromRaw() const
+{
+    return true;
+}
+
+std::string AssetLoaderLeaderboard::GetFileNameForAsset(const std::string& assetName)
+{
+    return std::format("leaderboards/{}.json", assetName);
+}
+
+bool AssetLoaderLeaderboard::LoadFromRaw(
+    const std::string& assetName, ISearchPath* searchPath, MemoryManager* memory, IAssetLoadingManager* manager, Zone* zone) const
+{
+    const auto file = searchPath->Open(GetFileNameForAsset(assetName));
+    if (!file.IsOpen())
+        return false;
+
+    auto* leaderboardDef = memory->Alloc<LeaderboardDef>();
+    leaderboardDef->name = memory->Dup(assetName.c_str());
+
+    if (LoadLeaderboardAsJson(*file.m_stream, *leaderboardDef, memory))
+        manager->AddAsset<AssetLeaderboard>(assetName, leaderboardDef);
+    else
+        std::cerr << "Failed to load leaderboard \"" << assetName << "\"\n";
```

Comment:
Since the codebase is now C++20 i'm now trying to make use of `std::format` in these case as much as possible.
For these kinds of places this also solves the issue of sometimes experiencing disconnected messages were one part of a log line would be behind another line (like you experienced in your runner).

It's also good preparation for a proper logging system that i kind of want to introduce at some point 😅